controllers/replication.storage/volumereplication_controller.go

```diff
@@ -363,16 +364,55 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	instance.Status.LastCompletionTime = getCurrentTime()
+
+	var requeueForInfo bool
+
+	if instance.Spec.ReplicationState == replicationv1alpha1.Primary {
+		info, err := r.getVolumeReplicationInfo(vr)
+		if err != nil {
+			logger.Error(err, "Failed to get volume replication info")
+			return ctrl.Result{}, err
+		}
+		lastSyncTime := info.GetLastSyncTime()
+
+		lastsynctime := metav1.NewTime(lastSyncTime.AsTime())
```

Comment:
very confusing naming, there is `lastSyncTime` and `lastsynctime`. If you really need both, make sure to call them  differently (maybe `ts` or `timestamp` and `lastSyncTime` or just `t`).